extensions/ipynb/src/notebookSerializer.ts

```diff
@@ -94,3 +93,17 @@ export class NotebookSerializer implements vscode.NotebookSerializer {
 		return JSON.stringify(notebookContent, undefined, indentAmount);
 	}
 }
+
+export function detectIndent(jsonString: string) {
```

Comment:
Forgot to complete this in the first submission of the PR.

Reply:
Though I like that we get rid of `detect-indent` a lot but I'm a bit worried if the algorithm here works the same as `detect-indent` for common cases. Not sure how much coverage we can have with tests 

Another algorithm to compare against is the one we have for monaco: https://github.com/microsoft/vscode/blob/79b66ea6ff73559c7481eb3ecde1cd3f5b4dd680/src/vs/editor/common/model/indentationGuesser.ts#L105-L1

Reply:
Not exactly sure what you;re asking here. Even if we had a mix of tabs & spaces this should work as `trimStart` would strip both of them.
And looking at the source you're provided, its ensuring spaces & tabs are both preserved.
E.g. if the json indentation is `\t    `, then we'll identify the tab+spaces & preserve that.

Reply:
I could optionally bring back the npm package & only provide the first 10-20 characters of the string. 
* That will address the issue with any edge cases
* Addresses perf issues (not providing the entire string, only part of it).

Thoughts?
@rebornix @roblourens /cc

Reply:
The problem IMHO is unformated document which contains both 2 spaces tab and 4 spaces tabs. 
The catch is indentation detection is always a `guess` and I think both `detect-indent` and monaco's one will detect the indentation of every line and find the most common indentation in the document. For example, if a document's first line uses 2 spaces but the rest of the document use 4 spaces, then the indentation from the algorithm would be 4, other than 2, while checking the first 20 characters will very likely give us 2.

In the core, we only detect indentation of the first 10,000 lines, maybe we can do the same for `detect-indent` (tested locally and it does help with the perf for large files).

Reply:
> contains both 2 spaces tab and 4 spaces tabs.
The catch is indentation detection is always a guess and I think both detect-indent

FYI - here we're not detecting indents entered by the user. 
We're detecting indents in the serialized JSON, not any user entered code. Hence your concerns are not really valid here. We're talking about the output of JSON.stringify (or equivalent in Python) - which is done by jupyter applicatino, not user input values